soundfile.py

```diff
@@ -467,6 +467,65 @@ def blocks(file, blocksize=None, overlap=0, frames=-1, start=0, stop=None,
             yield block
 
 
+class _SoundFileInfo(object):
+    """Information about a SoundFile"""
+    def __init__(self, file, verbose):
+        self.verbose = verbose
+        with SoundFile(file) as f:
+            self.name = f.name
+            self.samplerate = f.samplerate
+            self.channels = f.channels
+            self.duration = len(f)/f.samplerate
+            self.format = f.format
+            self.subtype = f.subtype
+            self.endian = f.endian
+            self.format_info = f.format_info
+            self.subtype_info = f.subtype_info
+            self.sections = f.sections
+            self.extra_info = f.extra_info
+
+    @property
+    def _duration_str(self):
+        hours = int(self.duration // 3600)
+        minutes = int((self.duration % 3600) // 60)
+        seconds = (self.duration % 60)
+        if hours >= 1:
+            duration = "{0}:{1:02}:{2:05.2f} h".format(hours, minutes, seconds)
+        elif minutes >= 1:
+            duration = "{0:02}:{1:05.2f} min".format(minutes, seconds)
+        else:
+            duration = "{0:.2f} s".format(seconds)
+        return duration
+
+    def __repr__(self):
+        info = "\n".join(
+            ["{0.name}",
+             "samplerate: {0.samplerate} Hz",
+             "channels: {0.channels}",
+             "duration: {0._duration_str}",
+             "format: {0.format_info} [{0.format}]",
+             "subtype: {0.subtype_info} [{0.subtype}]"])
+        if self.verbose:
+            info += "\n".join(
+                ["endian: {0.endian}",
+                 "sections: {0.sections}",
+                 'extra_info: """',
+                 '    {1}"""'])
+        indented_extra_info = ("\n"+" "*4).join(self.extra_info.split("\n"))
+        return info.format(self, indented_extra_info)
+
+
+def info(file, verbose=False):
+    """Returns an object with information about a SoundFile.
+
+    Parameters
+    ----------
+    verbose : bool
+        Whether to print additional information.
```

Comment:
Now that I'm reading this it seems strange to me that `verbose` only affects the `__repr__`.
But I guess that's the price we have to pay for combining human- and machine-readability, right?

Hmmm ...

Probably `verbose` should also affect the attributes of the `_SoundFileInfo` instance?
I guess this would be more consistent ...